bridge/src/xcm_asset_transactor.rs

```diff
@@ -0,0 +1,81 @@
+
+impl<CurrencyTransactor: TransactAsset, FungiblesTransactor: TransactAsset, AssetTypeChecker: AssetTypeIdentifier, Forwarder: TransactorForwarder> TransactAsset for XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder> {
+    // deposit_asset implements the TransactAsset deposit_asset method and contains the logic to classify
+    // the asset recipient location:
+    // 1. recipient is on the local parachain
+    // 2. recipient is on non-substrate chain(evm, cosmos, etc.)
+    // 3. recipient is on the remote parachain
+    fn deposit_asset(what: &MultiAsset, who: &MultiLocation, context: &XcmContext) -> XcmResult {
+        match (who.parents, who.first_interior()) {
+            // 1. recipient is on the local parachain
+            (0, Some(Parachain(_))) => {
+                // check if the asset is native, and call the corresponding deposit_asset()
+                if AssetTypeChecker::is_native_asset(what) {
+                    CurrencyTransactor::deposit_asset(what, who, context)?;
+                } else {
+                    FungiblesTransactor::deposit_asset(what, who, context)?
+                }
+            }
+            // recipient is on the remote chain
+            (1, Some(Parachain(_))) => {
+                // 2. recipient is on non-substrate chain(evm, cosmos, etc.), will forward to sygma bridge pallet
+                match who.interior {
+                    // sygma: 7379676d61000000000000000000000000000000000000000000000000000000
+                    // sygma-bridge: 7379676d612d6272696467650000000000000000000000000000000000000000
+                    X5(Parachain(1000), GeneralKey { length: 5, data: hex!["7379676d61000000000000000000000000000000000000000000000000000000"]},  GeneralKey { length: 12, data: hex!["7379676d612d6272696467650000000000000000000000000000000000000000"]}, GeneralIndex(..), GeneralKey { .. }) => {
+                        // check if the asset is native or foreign, and deposit the asset to a tmp account first
+                        let tmp_account = sp_io::hashing::blake2_256(&MultiLocation::new(0, X1(GeneralKey { length: 8, data: [2u8; 32] })).encode());
+                        if AssetTypeChecker::is_native_asset(what) {
+                            CurrencyTransactor::deposit_asset(&what.clone(), &Junction::AccountId32 { network: None, id: tmp_account }.into(), context)?;
+                        } else {
+                            FungiblesTransactor::deposit_asset(&what.clone(), &Junction::AccountId32 { network: None, id: tmp_account }.into(), context)?
+                        }
+
+                        Forwarder::other_world_transactor_forwarder(tmp_account, what.clone(), *who).map_err(|e| XcmError::FailedToTransactAsset(e.into()))?;
+                    }
+                    _ => {
+                        // 3. recipient is on remote parachain
```

Comment:
Forward to remote parachain is interesting, even though I don't know what exactly your dest spec would be like, but if it is something like `(1, X6[Parachain(your_parachain_id), Parachain(dest_parachain_id), ...]` being set on source parachain, then the pattern should be like `(0, X5(Parachain(dest_parachain_id), ...]`

Reply:
so my idea is to identify two location first: `local` and `outer`, outer means any location that matches with sygma bridge pattern, like other networks for example evm, cosmos, etc. Then the rest would be filled into the xcm bridge section, if xcm execution not success, it would raise an error and that means the provided location is invalid.